### PR TITLE
Add JSON file account import

### LINKED ISSUE #715: Allow importing geth-style json wallet files.

### PR DESCRIPTION
**Pull request by danfinlay**
There is now a menu item labeled "JSON File" for importing, and it can digest either:
- v1 MyEtherWallet JSON files
- v3 Account files (used by Geth, Mist, and MyEtherWallet).

Fixes #715

Also adds a new "loadingMessage" parameter to the appState, so we can indicate reasons for transitions that may take longer than usual (Geth hashes secret info so many times before encrypting it that it can take a long time to decrypt in a browser).

Also incorporates features previously mentioned in a PR I'm closing, #1028.  Those are:

Now any strategy for importing a private key that can be described as a pure function can be very easily turned into a MetaMask import strategy.

I've created a generic and reusable UI action called `importNewAccount(strategy, args)`.

The `strategy` is a unique string that maps to an import function in `app/scripts/account-import-strategies/index.js`, and the `args` will be passed to the member of the `strategies` array whose key matches the strategy string.

Strategies return private key hex strings, and are used by the metamask-controller to create a new keyring, and select that new account, before calling back.

These strategies simply need subviews made to trigger that action, which are being kept in `ui/app/accounts/import`, and they should be added to the import menu array in `ui/app/accounts/import/index.js`.

This also implements @frankiebee's idea of showing the imported account when it's been imported (my oversight!).

### CODE REVIEW

**Comment by flyswatter** [comment]
screen shot 2017-01-18 at 4 50 33 pm

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -121,6 +122,16 @@ module.exports = class MetamaskController extends EventEmitter {
         .then((newState) => { cb(null, newState) })
         .catch((reason) => { cb(reason) })
       },
+      importAccountWithStrategy: (strategy, args, cb) => {
+        accountImporter.importAccount(strategy, args)
+        .then((privateKey) => {
+          return keyringController.addNewKeyring('Simple Key Pair', [ privateKey ])
```
Comment (Zanibas) [inline app/scripts/metamask-controller.js]: I assume we'll add more logic here as we expand our import strategies

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -121,6 +122,16 @@ module.exports = class MetamaskController extends EventEmitter {
...
+        })
+        .then(keyring => keyring.getAccounts())
+        .then((accounts) => keyringController.setSelectedAccount(accounts[0]))
```
Comment (Zanibas) [inline app/scripts/metamask-controller.js]: So this selects the first account from the new accounts, not overall? Sounds good, just clarifying.

File: ui/app/actions.js [hunk 3]
```diff
@@ -264,6 +265,21 @@ function addNewKeyring (type, opts) {
   }
 }
 
+function importNewAccount (strategy, args) {
+  return (dispatch) => {
+    dispatch(actions.showLoadingIndication('This may take a while, be patient.'))
```
Comment (Zanibas) [inline ui/app/actions.js]: showLoadingIndication takes in messages now? Cool!

File: app/scripts/metamask-controller.js (hunk 1 above)
Comment (flyswatter) [inline app/scripts/metamask-controller.js]: This action is only really useful for importing individual accounts representing a single keypair.  We still have the `addNewKeyring` action for creating an arbitrary keyring (like HD, or whatever else we add).

File: app/scripts/metamask-controller.js (hunk 2 above)
Comment (flyswatter) [inline app/scripts/metamask-controller.js]: Yes, it gets the `accounts` from the `keyring` returned from `addNewKeyring`, which is the new one, and only ever has one account given those parameters.

File: ui/app/actions.js (hunk 3 above)
Comment (flyswatter) [inline ui/app/actions.js]: Yup!  Although this message will be a blur unless a person is importing a geth/mist wallet, but we can't know when we initialize it.

Comment (flyswatter) [inline ui/app/actions.js]: Actually maybe we could know, we could parse the JSON file for the number of hashes... /shrug...

**Review by 2-am-zzz — APPROVED** [review]

**Pull request merged**